domains/runtime/evm/src/lib.rs

```diff
@@ -908,49 +958,98 @@ impl_runtime_apis! {
             }
         }
 
-        fn check_transaction_validity(
+        fn check_transaction_validity(uxt: &<Block as BlockT>::Extrinsic,
+            block_number: BlockNumber,
+            block_hash: <Block as BlockT>::Hash) -> Result<(), domain_runtime_primitives::CheckTxValidityError>  {
+            let lookup = frame_system::ChainContext::<Runtime>::default();
+            check_transaction_validity_inner(&lookup, uxt, block_number, block_hash).map(|_| ())
+        }
+
+        fn check_transaction_validity_and_do_pre_dispatch(
             uxt: &<Block as BlockT>::Extrinsic,
             block_number: BlockNumber,
             block_hash: <Block as BlockT>::Hash
         ) -> Result<(), domain_runtime_primitives::CheckTxValidityError> {
             let lookup = frame_system::ChainContext::<Runtime>::default();
-            let maybe_signer_info = extract_signer_inner(uxt, &lookup);
-
-            if let Some(signer_info) = maybe_signer_info {
-                let signer = signer_info.map_err(|tx_validity_error| {
-                    domain_runtime_primitives::CheckTxValidityError::UnableToExtractSigner {
-                        error: tx_validity_error
-                    }
-                })?;
-
-                let tx_validity =
-                    Executive::validate_transaction(TransactionSource::External, uxt.clone(), block_hash);
 
-                tx_validity.map(|_| ()).map_err(|tx_validity_error| {
-                    domain_runtime_primitives::CheckTxValidityError::InvalidTransaction {
-                        error: tx_validity_error,
-                        storage_keys: storage_keys_for_verifying_tx_validity_inner(signer, block_number, Self::extrinsic_era(uxt)),
+            let maybe_signer = check_transaction_validity_inner(&lookup, uxt, block_number, block_hash)?;
+
+            let storage_keys = storage_keys_for_verifying_tx_validity_inner(maybe_signer, block_number, Self::extrinsic_era(uxt));
```

Comment:
I wonder if we still need to return these storage keys here, since the generation of fraud proof happens when there is mismatch of `ER::inbo_bundles` instead of when this check return error.